pkg/controller/replication_controller.go

```diff
@@ -220,6 +224,10 @@ func (rm *ReplicationManager) updatePod(old, cur interface{}) {
 	}
 	// TODO: Write a unittest for this case
 	curPod := cur.(*api.Pod)
+	if curPod.DeletionTimestamp != nil {
+		rm.deletePod(curPod)
```

Comment:
Should just serve as a signal to the RM that it may need to check this again (we observed a pod updated with a deletion timestamp) - I think we can only over check (wake up too early) but since each wake up also properly calls filterActivePods my understanding that post-deletion, we might wake up more times and do redundant checks.